Pokedex/ProfileViewController.swift

```diff
@@ -0,0 +1,296 @@
+            }else{
+                profilePic.image = UIImage(named: "noimage")
+                
+            }
+            
+        }else{
+            let imageName = "noimage"
+            let image = UIImage(named: imageName)
+            profilePic = UIImageView(image: image!)
+            
+        }
+
+    
+        profilePic.contentMode = .scaleAspectFit
+        profilePic.clipsToBounds = true
+        containerView.addSubview(profilePic)
+        
+        // Instantiates the class that displays the Pokemon's name and number, the two most important properties.
+        nameAndNumberView = NameAndNumberView(frame: CGRect(x: view.frame.width * 0.25, y: (view.frame.height * 0.3 + 30) + self.navigationController!.navigationBar.frame.height, width: view.frame.width * 0.5, height: 55), pokemon: passedPokemon!)
+        containerView.addSubview(nameAndNumberView!)
+        
+        // Instantiates the class that displays the Pokemon's other properties/statistics.
+        let otherStatisticsView = OtherStatisticsView(frame: CGRect(x: view.frame.width * 0.15, y: (view.frame.height * 0.3 + 105) + self.navigationController!.navigationBar.frame.height, width: view.frame.width * 0.7, height: 257.5), pokemon: passedPokemon!)
+        containerView.addSubview(otherStatisticsView)
+        
+        // Instantiates the class that contains the ADDTOFAVORITES button.
+        let addToFavorites = AddToFavorites(frame: CGRect(x: view.frame.width * 0.15, y: (view.frame.height * 0.3 + 382.5) + self.navigationController!.navigationBar.frame.height, width: view.frame.width * 0.7, height: 40), staticText: "Add To Favorites")
+        containerView.addSubview(addToFavorites)
+        
+    }
+    
+    /*
+     // MARK: - Navigation
+     
+     // In a storyboard-based application, you will often want to do a little preparation before navigation
+     override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+     // Get the new view controller using segue.destinationViewController.
+     // Pass the selected object to the new view controller.
+     }
+     */
+    
+}
+
+
+class NameAndNumberView: UIView {
+    
+    var pokemon: Pokemon!
+    
+    // Takes in the POKEMON object as a parameter.
+    init(frame: CGRect, pokemon: Pokemon) {
+        super.init(frame: frame)
+        self.backgroundColor = UIColor.white
+        self.layer.cornerRadius = 5
+        self.clipsToBounds = true
+        self.pokemon = pokemon
+        setNameNumLayout()
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    func setNameNumLayout() {
+        
+        // Sets the layout for the Pokemon's NAME.
+        let nameLabel = UILabel(frame: CGRect(x: frame.width * 0.1, y: 5, width: frame.width * 0.8, height: 20))
+        nameLabel.text = "Name: \(pokemon.name!)"
+        nameLabel.font = UIFont(name: "HelveticaNeue-SemiBold", size: 16.0)
```

Comment:
Sometimes your complete name is not shown and may say: Name: Pika...

To avoid the ellipsis, 

factLabel.numberOfLines = 1;
factLabel.minimumFontSize = 8;
factLabel.adjustsFontSizeToFitWidth = YES;